Source/reactions/Castro_react.cpp

```diff
@@ -139,32 +139,56 @@ Castro::react_state(MultiFab& s, MultiFab& r, Real time, Real dt)
 
                 if (do_burn) {
 
-                    // Add burning rates to reactions MultiFab, but be
-                    // careful because the reactions and state MFs may
-                    // not have the same number of ghost cells.
+                    // update the state
 
-                    if (reactions.contains(i,j,k)) {
+                    if (U.contains(i,j,k)) {
                         for (int n = 0; n < NumSpec; ++n) {
-                            reactions(i,j,k,n) = U(i,j,k,URHO) * (burn_state.xn[n] - U(i,j,k,UFS+n) * rhoInv) / dt;
+                            U(i,j,k,UFS+n) += U(i,j,k,URHO) * (burn_state.xn[n] - U(i,j,k,UFS+n) * rhoInv);
                         }
 #if NAUX_NET > 0
                         for (int n = 0; n < NumAux; ++n) {
-                            reactions(i,j,k,n+NumSpec) = U(i,j,k,URHO) * (burn_state.aux[n] - U(i,j,k,UFX+n) * rhoInv) / dt;
+                            U(i,j,k,UFX+n) += U(i,j,k,URHO) * (burn_state.aux[n] - U(i,j,k,UFX+n) * rhoInv);
                         }
 #endif
-                        reactions(i,j,k,NumSpec+NumAux  ) = U(i,j,k,URHO) * burn_state.e / dt;
-                        reactions(i,j,k,NumSpec+NumAux+1) = amrex::max(1.0_rt, static_cast<Real>(burn_state.n_rhs + 2 * burn_state.n_jac));
+                        U(i,j,k,UEINT) += U(i,j,k,URHO) * burn_state.e;
+                        U(i,j,k,UEDEN) += U(i,j,k,URHO) * burn_state.e;
+                    }
+
+                    // Add burning rates to reactions MultiFab, but be
+                    // careful because the reactions and state MFs may
+                    // not have the same number of ghost cells.
+
+                    if (reactions.contains(i,j,k)) {
+
+                        reactions(i,j,k,0) = U(i,j,k,URHO) * burn_state.e / dt;
+                        reactions(i,j,k,1) = amrex::max(1.0_rt, static_cast<Real>(burn_state.n_rhs + 2 * burn_state.n_jac));
+
+                        if (store_omegadot == 1) {
+                            if (reactions.contains(i,j,k)) {
+                                for (int n = 0; n < NumSpec; ++n) {
+                                    reactions(i,j,k,2+n) = U(i,j,k,URHO) * (burn_state.xn[n] - U(i,j,k,UFS+n) * rhoInv) / dt;
```

Comment:
This needs to be done before the update to U, otherwise since U has already been updated to be the same as burn_state.xn this will just be zero

Reply:
fixed